.jenkins/pytorch/macos-build.sh

```diff
@@ -12,12 +12,14 @@ if [ -z "${IN_CI}" ]; then
   export DEVELOPER_DIR=/Applications/Xcode9.app/Contents/Developer
 fi
 
-if which sccache > /dev/null; then
-  printf "#!/bin/sh\nexec sccache %s \$*" "$(which clang++)" > "${WORKSPACE_DIR}/clang++"
-  chmod a+x "${WORKSPACE_DIR}/clang++"
+function write_sccache_stub() {
+  printf "#!/bin/sh\nif [ \$(ps auxc \$(ps auxc -o ppid \$\$ | grep \$\$ | rev | cut -d' ' -f1 | rev) | tr '\\\\n' ' ' | rev | cut -d' ' -f2 | rev) != sccache ]; then\n  exec sccache $(which $1) \"\$@\"\nelse\n  exec $(which $1) \"\$@\"\nfi" > "${WORKSPACE_DIR}/$1"
```

Comment:
this command is absolutely gross, but I'm not sure how to fix it (I tried using pgrep but the command wasn't equivalent/take in a PID). 

This is what it looks like in the script:
`ps auxc $(ps auxc -o ppid $$ | grep $$ | rev | cut -d' ' -f1 | rev) | tr '\n' ' ' | rev | cut -d' ' -f2 | rev`